--- a/src/kexecboot.c	2023-12-24 23:33:28.176826166 +0200
+++ b/src/kexecboot.c	2023-12-24 23:36:35.996053907 +0200
@@ -378,10 +378,10 @@
 		exit(-1);
 	}
 
-	umount(mount_point);
-
 	dispose(cmdline_arg);
 
+	/* Only umount the read-only mount_point if kexec fails, see below */
+
 	/* Check /proc/sys/net presence */
 	if ( -1 == stat("/proc/sys/net", &sinfo) ) {
 		if (ENOENT == errno) {
@@ -399,6 +399,8 @@
 	/* Boot new kernel */
 	execve(exec_argv[0], (char *const *)exec_argv, envp);
 
+	umount(mount_point);
+
 free:
 	dispose(cmdline_arg);
 	for (idx = 0; idx < MAX_LOAD_ARGV_NR; idx++)
